network/ServerNetworking.cpp

```diff
@@ -464,26 +466,51 @@ void PlayerConnection::AsyncReadMessage() {
                                         boost::asio::placeholders::bytes_transferred));
 }
 
-void PlayerConnection::SyncWriteMessage(const Message& message) {
-    // Synchronously write and asynchronously signal the errors.  This prevents PlayerConnections
-    // being removed from the list while iterating to transmit to multiple receivers.
-    Message::HeaderBuffer header_buf;
-    HeaderToBuffer(message, header_buf);
-    std::vector<boost::asio::const_buffer> buffers;
-    buffers.push_back(boost::asio::buffer(header_buf));
-    buffers.push_back(boost::asio::buffer(message.Data(), message.Size()));
+void PlayerConnection::SendMessageImpl(PlayerConnectionPtr self, Message message) {
+    bool start_write = self->m_outgoing_messages.empty();
```

Comment:
could you comment to document what this function does / why?

Reply:
I've added some documentation bit I amn't sure why, I've just copied it from `ClientNetworking.cpp`. I suppose it related to multi-threading.